### ISSUE TITLE
Contact names not showing on device

### ISSUE THREAD

**Issue by justinriddoch** [shusain93/Andromeda-iMessage#16]
I just got everything up and running, and it's working great other than on my device there are no contact names. I am getting the following error though: 

2017-10-26 22:51:31.760 MessageProxy[2391:54645] [Accounts] Failed to update account with identifier <redacted>, error: Error Domain=ABAddressBookErrorDomain Code=1002 "(null)"

Any ideas what could be causing it? I am using a gmail account in the contacts app for all of the contacts if that makes a difference.

**Comment by ezhes** [comment]
Contacts are actually pulled from your Mac. Are you signed in properly there? Did you grant contact access?

**Comment by justinriddoch** [comment]
Yeah the contacts app on Mac has my gmail account signed in, would I have to copy the contacts to 'All on my Mac' section? The contact names show in the iMessage app by the way.

**Comment by ezhes** [comment]
No idea, I assume iMessage has working contacts? Can you try launching it without the script?

**Comment by justinriddoch** [comment]
I got an error saying the app was corrupt when I tried launching it, so I redownloaded it and now some contacts are showing. I am testing with one phone number and one email, and the phone number works. Will it pull emails contact names?

Edit: When I switched the contact I was texting with from an email to phone number it worked, so I guess email contact names aren't supported?

**Comment by ezhes** [comment]
Email contact names are supported, not sure why it's not working for you. If you copy and paste the exact email into your macs contacts app search, do you get the contact back? Is it on the same account as your other ones? Is it an exact match? It's sort of sketch-ily done honestly but it does work, I assure you.

**Comment by justinriddoch** [comment]
Unfortunately that didn't fix it, I'm going to try restarting everything and see if that helps.

It is the same account as the other one, saved under the same contact name.

**Comment by ezhes** [comment]
That wasn't it a fix, it was a question. The app is doing a direct matching for contact names and if it's not exactly the same (ie there's a space or something at the end, I don't know) it won't work. Can you make a "on my mac" contact with just the email address in it to test if it's an issue with it being on google?

**Comment by justinriddoch** [comment]
I tried importing the contact from gmail and creating a new contact with just the email address, but the contact name isn't showing. However, I think I am going to have to test it with a different contact because now when I try to message the email, the messages transfer over into the phone number convo and doesn't appear in the email comvo. My guess is that after I started texting the phone number that the email convo stopped updating. Does that make any sense?
